api/v1/sriovoperatorconfig_types.go

```diff
@@ -24,6 +24,15 @@ type SriovOperatorConfigSpec struct {
 	DisableDrain bool `json:"disableDrain,omitempty"`
 	// Flag to enable OVS hardware offload. Set to 'true' to provision switchdev-configuration.service and enable OpenvSwitch hw-offload on nodes.
 	EnableOvsOffload bool `json:"enableOvsOffload,omitempty"`
+	// OvsHardwareOffload describes the OVS HWOL configuration for selected MachineConfigPool
+	OvsHardwareOffload []OvsHardwareOffloadConfig `json:"ovsHardwareOffload,omitempty"`
```

Comment:
I haven't seen such an API design in K8S API.  Normally, if users need to select objects, they shall use a selector which matches the label of the objects. Personally, I think this implementation is hard to understand.

Shall we ask users to label the MCP by themself, and we use a selector here?

Reply:
> I haven't seen such an API design in K8S API. Normally, if users need to select objects, they shall use a selector which matches the label of the objects. Personally, I think this implementation is hard to understand.

My previous thinking was that we may want to re-use the same API for k8s case in the future.
Current `OvsHardwareOffload.MachineConfigPoolSelector` matches on node labels, where for OpenShift, it matches the (node) labels defined in MCP `nodeSelector`, for k8s, it could match the node labels directly.
> 
> Shall we ask users to label the MCP by themself, and we use a selector here?

One idea I had before was to label or annotate MCP (with ovs-hw-offload) directly, then not change the SR-IOV API, instead, we can have SR-IOV Operator watch on MCP update for creating/deleting HWOL MCs. wdyt?

Reply:
> > I haven't seen such an API design in K8S API. Normally, if users need to select objects, they shall use a selector which matches the label of the objects. Personally, I think this implementation is hard to understand.
> 
There is a similar example in tuned Operator, which was suggested by MCO team: https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/apis/tuned/v1/tuned_types.go#L79
It defines the MachineConfigLabels in Tuned API to match on MCPs.

Reply:
> My previous thinking was that we may want to re-use the same API for k8s case in the future.

If we plan to reuse it for k8s, the name `MachineConfigPoolSelector` is kind of misleading. How about just use `NodeSelector` instead?

> One idea I had before was to label or annotate MCP (with ovs-hw-offload) directly, then not change the SR-IOV API, instead, we can have SR-IOV Operator watch on MCP update for creating/deleting HWOL MCs. wdyt?

I prefer we can have a common API for both OCP and K8S. Otherwise, we still need a way to tell when the ovs hw offload shall be applied in the k8s case. Currently, it's hard bonded with the switchdev configuration.

Reply:
Changed to `NodeSelector`